File: src/main/java/org/fretron/kafka/serializer/CustomSerializer.java
package org.fretron.kafka.serializer;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.apache.kafka.common.serialization.Serializer;

import java.util.Map;

public class CustomSerializer implements Serializer<Object> {
    @Override
    public void configure(Map configs, boolean isKey) {
        // TODO: 11/12/20
    }

    @Override
    public byte[] serialize(String topic, Object data) {
        byte[] retVal = null;
        ObjectMapper objectMapper = new ObjectMapper();
        try {
            retVal = objectMapper.writeValueAsString(data).getBytes();
        } catch (Exception e) {
            e.printStackTrace();
        }
        return retVal;
    }

    @Override
    public void close() {
        // TODO: 11/12/20
    }
}

File: src/main/java/org/fretron/kafka/serializer/CustomSerializer.kt
package org.fretron.kafka.serializer

import com.fasterxml.jackson.databind.ObjectMapper
import org.apache.kafka.common.serialization.Serializer

class CustomSerializer : Serializer<Any> {

    override fun serialize(topic: String?, data: Any): ByteArray {
        var retVal: ByteArray? = null
        val objectMapper = ObjectMapper()
        try {
            retVal = objectMapper.writeValueAsString(data).toByteArray()
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return retVal!!
    }

}